### DISCUSSION TITLE
Auto-update Tailwind styling

### DISCUSSION THREAD

**Discussion by natankeddem** [zauberzeug/nicegui#2119] [Ideas / Feature Requests]
### Description

I want to start off saying I think this will probably require a significant a effort to solve. So, I don't know if you will want to change the current functionality. Perhaps we could document this behavior a bit better and that would be good enough.

Right now using `tailwind` works well for static styling. I feel it doesn't work as expected when you try to use it in a dynamic way. For example changing background colors on the fly.

```py
from nicegui import ui
from typing import Literal

status = ui.label("Press a Button!")

def mode(mode: Literal["pass", "fail", "busy"]) -> None:
    color = {"pass": "green", "fail": "red", "busy": "orange"}
    status.text = mode
    status.tailwind.background_color(color[mode])

ui.button("PASS", on_click=lambda _: mode("pass"))
ui.button("FAIL", on_click=lambda _: mode("fail"))
ui.button("BUSY", on_click=lambda _: mode("busy"))

ui.run()
```

I understand why the background colors don't change correctly here. `tailwind` is just adding multiple `bg-color` classes without removing the old `bg-color`. In my opinion it would be nice if the old `bg-color` was removed and then a new `bg-color` added in its place. This would require some way of keeping track of tailwind selections and applying them similarly to `props`. Also, I am not if certain `tailwind` methods might need more than 1 selection in certain situations which would throw a wrench into this plan.

**Comment by falkoschindler** [comment]
> This would require some way of keeping track of tailwind selections and applying them similarly to props. Also, I am not if certain tailwind methods might need more than 1 selection in certain situations which would throw a wrench into this plan.

I like the general idea, but we should clarify these points before attempting to implement this feature.
